tests/integration/common.go

```diff
@@ -607,3 +613,18 @@ func (s *CCVTestSuite) setupValidatorPowers() {
 	}
 	s.Require().Equal(int64(4000), stakingKeeper.GetLastTotalPower(s.providerCtx()).Int64())
 }
+
+// mustGetStakingValFromTmVal returns the staking validator from the current state of the staking keeper,
+// corresponding to a given tendermint validator. Note this func will fail the test if the validator is not found.
+func (s *CCVTestSuite) mustGetStakingValFromTmVal(tmVal tmtypes.Validator) (stakingVal stakingtypes.Validator) {
+	vals := s.providerApp.GetTestStakingKeeper().GetAllValidators(s.providerCtx())
```

Comment:
Why are we iterating over all validators instead of getting the validator by its consensus address here?
i.e. `stakingKeeper.GetValidatorByConsAddr(sdk.ConsAddress(tmVal.Address))`

Reply:
Great catch, fixed!